OOP/LibraryManagement/LibraryApp/Repositories/JsonFileDocumentRepository.cs

```diff
@@ -0,0 +1,99 @@
+﻿using LibraryApp.Documents;
+using Newtonsoft.Json.Linq;
+using System.Text.RegularExpressions;
+using System.Text.Json;
+using LibraryApp.Services;
+
+namespace LibraryApp.Repositories;
+
+public class JsonFileDocumentRepository<T>(DocumentCache cache) : IRepository<T> where T : Document
+{
+    private const string DocType = "documentType";
+    private readonly DocumentCache _cache = cache;
+    private readonly string _baseDirectory = AppDomain.CurrentDomain.BaseDirectory;
```

Comment:
In general, I would advise to pass `baseDirectory` as an argument via constructor params - it will make repository more reusable and flexible if base location of file will be changed